sympy/solvers/solveset.py

```diff
@@ -978,6 +985,331 @@ def _solveset(f, symbol, domain, _check=False):
+    equation can be easily handled by `solveset`.
+
+    For example:
+
+    .. math:: 3^{2x} = 2^{x + 3}
+
+    Taking log both sides will reduce the equation to
+
+    .. math:: (2x)*log(3) = (x + 3)*log(2)
+
+    This form can be easily handed by `solveset`.
+    """
+
+    if not (isinstance(f, Add) and len(f.args) == 2):
+        # solving for the sum of more than two powers is possible
+        # but not yet implemented
+        return None
+
+    lhs, rhs = list(ordered(f.args))
+
+    log_type_equation = expand_log(log(lhs)) - expand_log(log(-rhs))
+
+    return log_type_equation
+
+
+def _check_expo(f, symbol):
+    r"""
+    Helper to check whether an equation is exponential or not.
+
+    ``f`` is the equation to be checked
+
+    ``symbol`` is the variable in which the equation needs to be checked
+
+    The function extracts each term of the equation and checks if it is
+    of exponential form w.r.t `symbol`.
+    Returns `True` if any term is of exponential type otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, cos, exp
+    >>> from sympy.solvers.solveset import _check_expo as check
+    >>> x, y = symbols('x y')
+    >>> check(x**y - x, y)
+    True
+    >>> check(x**y - x, x)
+    False
+    >>> check(exp(x + 3), x)
+    True
+    >>> check(cos(2**x), x)
+    False
+    """
+
+    for add_arg in Add.make_args(f):
+        for mul_arg in Mul.make_args(add_arg):
+            if isinstance(mul_arg, (Pow, exp)) and \
+                    symbol in mul_arg.exp.free_symbols:
+                return True
+    return False
+
+
+def _transolve(f, symbol, domain):
+    r"""
+    Function to solve transcendental equations. It is a helper to
+    `solveset` and should be used internally. `\_transolve`
+    currently supports the following class of equations:
+
+        - Exponential equations
+
+    Parameters
+    ==========
+
+    f : is any transcendental equation that needs to be solved.
+        This needs to be an expression, which is assumed
+        to be equal to 0.
+
+    symbol : is variable for which the equation is solved.
+        This needs to be of class `Symbol`.
+
+    domain : is a set over which the equation is solved.
+        This needs to be of class `Set`.
+
+    How to use `\_transolve`
+    ========================
+
+    `\_transolve` should not be used as an independent function, because
+    it assumes that the equation (`f`) and the `symbol` comes from
+    `solveset` and might have undergone a few modification(s).
+    To use `\_transolve` as an independent function the equation (`f`)
+    and the `symbol` should be passed as they would have been by
+    `solveset`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _transolve as transolve
+    >>> from sympy.solvers.solvers import _tsolve as tsolve
+    >>> from sympy import symbols, S, pprint
+    >>> # symbol modified as would have been by solveset
```

Comment:
When solveset calls transolve it will pass assumptions with
the symbols. For demonstration purposes, we need to do so,
too.

>>> x = symbols('x', real=True)